mdsuite/calculators/structure_factor.py

```diff
@@ -250,7 +236,10 @@ def average_atomic_form_factor(self, scattering_scalar):
         sum1 = 0
         atomic_form_facs = self.atomic_form_factors(scattering_scalar)
         for el in self.experiment.species:
-            sum1 += self.molar_fractions[el] * atomic_form_facs[el]["atomic_form_factor"]
+            sum1 += (
+                self.elements_dict[el]["molar_fraction"]
+                * atomic_form_facs[el]["atomic_form_factor"]
+            )
```

Comment:
This can probably be rewritten as `sum([element["molar_fraction"] * form_fac['atomic_form_factor'] for element, form_fac in zip(self.element_dict.values(), atomic_form_facs.values())])`

but more important, why is `atomic_form_facs` a nested dict of `{species{"atomic_form_factor": value}}`? Shouldn't it be `{species: value}` ? Furthermore, you use "molar_fraction" a lot, it might be easier to create a `self.molar_fraction = {species: fraction}` and then the sum above would simplify a lot.

Reply:
I like to store all the information in one dict, I also think it read quite nicely actually. More oversightful if all the informatin is stored in one place.

Reply:
For the structure factor you need all the rdfs anyways, there is no case where you just choose some of the rdfs.

I think the storing is fine, why shouldnt you have one experiment.species dict where everything is stored. If it is changed its just going to break a lot of stuff again and I dont see an advantage.